drivers/common/openserial.c

```diff
@@ -282,6 +282,162 @@ uint8_t openserial_getInputBuffer(uint8_t* bufferToWrite, uint8_t maxNumBytes) {
     
     return numBytesWritten;
 }
+
+//===== Statistics
+
+owerror_t openserial_printStat(uint8_t type, uint8_t *buffer, uint8_t length) {
+#ifdef OPENSERIAL_STAT
+    uint8_t i;
+    uint8_t  asn[5];
+    
+    // retrieve ASN
+    INTERRUPT_DECLARATION();
+    ieee154e_getAsn(asn);
+    
+    DISABLE_INTERRUPTS();
+    openserial_vars.outputBufFilled  = TRUE;
+    outputHdlcOpen();
+    outputHdlcWrite(SERFRAME_MOTE2PC_STAT);
+    outputHdlcWrite(idmanager_getMyID(ADDR_16B)->addr_16b[0]);
+    outputHdlcWrite(idmanager_getMyID(ADDR_16B)->addr_16b[1]);
+    for(i=0;i<5;i++)
+    		outputHdlcWrite(asn[i]);
+    outputHdlcWrite(type);
+    for (i=0;i<length;i++){
+        outputHdlcWrite(buffer[i]);
+    }
+    outputHdlcClose();
+    ENABLE_INTERRUPTS();
+#endif
+    
+    return E_SUCCESS;
+}
+
+
+//===== Statistics - Schedule
+
+
+//a cell was inserted in the schedule
+void openserial_statCell(uint8_t code, scheduleEntry_t* slotContainer){
+#ifdef OPENSERIAL_STAT
+    evtCell_t evt;
+    evt.code          = code;
+    evt.slotOffset      = slotContainer->slotOffset;
+    evt.type            = slotContainer->type;
+    evt.shared          = slotContainer->shared;
+    evt.channelOffset   = slotContainer->channelOffset;
+    memcpy(evt.neighbor,      slotContainer->neighbor.addr_64b, 8);
+    openserial_printStat(SERTYPE_CELL, (uint8_t*)&evt, sizeof(evt));
+#endif
+}
+
+
+//===== Statistics - Packets
+
+//a ack was txed or rcvd
+void openserial_statAck(uint8_t code, open_addr_t *l2addr){
+#ifdef OPENSERIAL_STAT
+    evtAck_t evt;
+    evt.code = code;
+    memcpy(evt.l2addr,      l2addr->addr_64b, 8);
+    openserial_printStat(SERTYPE_ACK, (uint8_t*)&evt, sizeof(evt));
+#endif
+}
+
+//push an event to track received frames
+void openserial_statRx(OpenQueueEntry_t* msg){
+#ifdef OPENSERIAL_STAT
+   evtPktRx_t evt;
+   evt->length                      = msg->length;
+   evt->rssi                        = msg->l1_rssi;
+   evt->lqi                         = msg->l1_lqi;
+   evt->crc                         = msg->l1_crc;
+   evt->frame_type                  = msg->l2_frameType;
+   evt->slotOffset                  = schedule_getSlotOffset();
+   evt->frequency                   = calculateFrequency(schedule_getChannelOffset());
+   memcpy(evt->l2Src,   msg->l2_nextORpreviousHop.addr_64b, 8);
+   openserial_printStat(SERTYPE_PKT_RX, (uint8_t*)&evt, sizeof(evtPktRx_t));
+#endif
+}
+
+//push an event to track transmitted frames
+void openserial_statTx(OpenQueueEntry_t* msg){
+#ifdef OPENSERIAL_STAT
+   evtPktTx_t evt;
+   evt->length                      = msg->length;
+   evt->txPower                     = msg->l1_txPower;
+   evt->numTxAttempts               = msg->l2_numTxAttempts;
+   evt->frame_type                  = msg->l2_frameType;
+   evt->slotOffset                  = schedule_getSlotOffset();
+   evt->frequency                   = calculateFrequency(schedule_getChannelOffset());
+   
```

Comment:
As you are printing "L4Proto={12}|L4SrcPort={13}|L4DestPort={14}" in  LogPktTx() function, need to fill them here.